### ISSUE TITLE
No data

### ISSUE THREAD

**Issue by fuuman** [xavier-hernandez/goaccess-for-nginxproxymanager#31]
```
version: "3.9"
services:
  goaccess:
    image: justsky/goaccess-for-nginxproxymanager:latest
    container_name: goaccess
    restart: always
    environment:
        - TZ=Europe/Berlin
        - SKIP_ARCHIVED_LOGS=False #optional
        - BASIC_AUTH=False #optional
        - BASIC_AUTH_USERNAME=user #optional
        - BASIC_AUTH_PASSWORD=pass #optional
    ports:
        - '7880:7880'
    volumes:
        - /data/nginx-proxy-manager/app/data/logs:/opt/log
```
I am using the `docker-compose.yml` above and under `/data/nginx-proxy-manager/app/data/logs` are multiple log files (`proxy_host-x.log`) files. I see zero data. Everything is 0. Just `Log Parsing Time` is 1 second. So I cannot see any old or live data. What am I doing wrong?

**Comment by xavier-hernandez** [comment]
Someone else just posted the same thing...

I recommend you do "docker container logs goaccess" it should tell you if the docker container can read your logs files. It could be a permission issue.

**Comment by fuuman** [comment]
I don't think that it has anything to do with permissions. And the log just tells me "no logs to scan". The problem is that npm is saving its files like I mentioned above. As `proxy_host-x.log`. And goaccess looks for `proxy_host-x_access.log`. That's the whole problem.

**Comment by xavier-hernandez** [comment]
@fuuman what version of NPM are you running?

I just pushed out a dev docker image to address this yesterday for another user. Can you try that? 

xavierh/goaccess-for-nginxproxymanager:develop

**Comment by fuuman** [comment]
The newest, which is `v2.9.16`.

**Comment by xavier-hernandez** [comment]
I'm running the same version and I'm not sure why there is a difference in file names. It's frustrating. I was hoping you were running 2.18 and at some point, there was a change in the filename structure. So that thought is out the window.

@fuuman you'd have to wait for @Just5KY to update his images

When I have time I'll try to implement the ARM images as well on my end.

**Comment by fuuman** [comment]
Which docker image of npm are you using?

**Comment by xavier-hernandez** [comment]
jc21/nginx-proxy-manager:latest

**Comment by fuuman** [comment]
Okay, that's sick, I am actually using the exact same image. :D 

I cannot believe that your names are named different. Thats the result of `ls`in my npm log folder: 
```
default.log               proxy_host-10.log  proxy_host-1.log  proxy_host-6.log
error.log                 proxy_host-11.log  proxy_host-2.log  proxy_host-7.log
fallback_access.log       proxy_host-12.log  proxy_host-3.log  proxy_host-8.log
fallback_error.log        proxy_host-13.log  proxy_host-4.log  proxy_host-9.log
letsencrypt-requests.log  proxy_host-14.log  proxy_host-5.log
```
Your files have `_access`? Then the only thing could be the architecture, because I am using the ARM image and you don't.

**Comment by xavier-hernandez** [comment]
Here is a sample...

![image](https://user-images.githubusercontent.com/6011223/159565515-056ec00e-bc43-4ffb-aa4e-836475f1a986.png)

**Comment by fuuman** [comment]
Okay, I have no clue.. super weird.

**Comment by fuuman** [comment]
@xavier-hernandez Another weird thing is that I have zero gz files. Maybe it's a configuration issue but I ccanot find any npm documentation according to logging.

**Comment by Just5KY** [comment]
My image is updated

**Comment by xavier-hernandez** [comment]
@Just5KY - no problem. Yeah, I'm trying to solve this filename issue, with 2 users, so I just committed to the develop branch. Thanks for your help!

**Comment by Just5KY** [comment]
Swear, this goes faster when I'm not looking
![image](https://user-images.githubusercontent.com/71321862/159579578-0f16aba5-1434-4f17-8e7c-f479b2f657d1.png)

**Comment by xavier-hernandez** [comment]
@Just5KY sorry bud. I hope to take this off your plate soon. I finally dug up my old raspberry PI so that I can test out at least one of the builds for ARM.

**Comment by fuuman** [comment]
I am sorry to say that but it's not working for me. I just pulled your new develop image. But just zeros in the frontend. Maybe I have another error. Because even when I do `cp proxy-host_1.log proxy-host_1_access.log` in my npm log folder and restart the goaccess container I have the exact same result...

**Comment by Just5KY** [comment]
Container logs still the same?

**Comment by Just5KY** [comment]
This is  the naming convention in my NPM logs
![image](https://user-images.githubusercontent.com/71321862/159583889-05c91cc1-97af-441b-acae-35333eebcd93.png)

**Comment by xavier-hernandez** [comment]
@fuuman I assume you meant "proxy-host-1.log" and not "proxy-host_1.log", typo? I'd recommend you enter the container and see if it has access to the files. 

"docker exec -it goaccess /bin/bash" then navigate to /opt/log and check to make sure the files are there and are readable. 

Thanks.

**Comment by xavier-hernandez** [comment]
> Container logs still the same?

No changes to anything other than modifying the script that finds the proxy files. 

https://github.com/xavier-hernandez/goaccess-for-nginxproxymanager/commit/e773f462c571fa08777d5e294d1c57672ff1f2fa

**Comment by fuuman** [comment]
Yes, typo. 

Still the exact container log for goaccess.

Yes, I was in the container. Logs are there and readable. Can open them.

**Comment by Just5KY** [comment]
@fuuman can you share your docker-compose for nginxproxy

```yml
  npm:
    container_name: npm_app
    image: 'jc21/nginx-proxy-manager:latest'
    restart: unless-stopped
    networks:
      - nginx
    ports:
      - '80:80' # Public HTTP Port
      - '443:443' # Public HTTPS Port
      - '81:81' # Admin Web Port
    environment:
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: 
      DB_MYSQL_PASSWORD:
      DB_MYSQL_NAME: "npm"
    volumes:
      - $DOCKERDIR/nginxproxy/data:/data
      - $DOCKERDIR/nginxproxy/letsencrypt:/etc/letsencrypt
    depends_on:
      - db
```

**Comment by fuuman** [comment]
```
version: "3"
services:
  app:
    image: 'jc21/nginx-proxy-manager:latest'
    container_name: nginx-proxy-manager_app
    restart: always
    ports:
      # Public HTTP Port:
      - '80:80'
      # Public HTTPS Port:
      - '443:443'
      # Admin Web Port:
      - '81:81'
    environment:
      # These are the settings to access your db
      DB_MYSQL_HOST: "db"
      DB_MYSQL_PORT: 3306
      DB_MYSQL_USER: "npm"
      DB_MYSQL_PASSWORD: "xxx"
      DB_MYSQL_NAME: "npm"
    volumes:
      - /home/pi/docker_data/nginx-proxy-manager/app/data:/data
      - /home/pi/docker_data/nginx-proxy-manager/app/letsencrypt:/etc/letsencrypt
    depends_on:
      - db
    networks:
      - firefly
      - nginx
      - bitwarden
  db:
    image: 'jc21/mariadb-aria:latest'
    container_name: nginx-proxy-manager_db
    restart: always
    environment:
      MYSQL_ROOT_PASSWORD: 'yyy'
      MYSQL_DATABASE: 'npm'
      MYSQL_USER: 'npm'
      MYSQL_PASSWORD: 'xxx'
    volumes:
      - /home/pi/docker_data/nginx-proxy-manager/db:/var/lib/mysql
    networks:
      - nginx
networks:
  nginx:
    name: nginx
  firefly:
    name: firefly
  bitwarden:
    name: bitwarden
```

**Comment by Just5KY** [comment]
Its exactly same

**Comment by xavier-hernandez** [comment]
Shouldn't your volume be like this?

version: "3.9"
services:
  goaccess:
    image: justsky/goaccess-for-nginxproxymanager:latest
    container_name: goaccess
    restart: always
    environment:
        - TZ=Europe/Berlin
        - SKIP_ARCHIVED_LOGS=False #optional
        - BASIC_AUTH=False #optional
        - BASIC_AUTH_USERNAME=user #optional
        - BASIC_AUTH_PASSWORD=pass #optional
    ports:
        - '7880:7880'
    volumes:
        **- /home/pi/docker_data/nginx-proxy-manager/app/data/logs:/opt/log**

**Comment by Just5KY** [comment]
You did change the volume to your nginx proxy log location, right? @fuuman 

> ```
> version: "3.9"
> services:
>   goaccess:
>     image: justsky/goaccess-for-nginxproxymanager:latest
>     container_name: goaccess
>     restart: always
>     environment:
>         - TZ=Europe/Berlin
>         - SKIP_ARCHIVED_LOGS=False #optional
>         - BASIC_AUTH=False #optional
>         - BASIC_AUTH_USERNAME=user #optional
>         - BASIC_AUTH_PASSWORD=pass #optional
>     ports:
>         - '7880:7880'
>     volumes:
>         - /data/nginx-proxy-manager/app/data/logs:/opt/log
> ```
> 
> I am using the `docker-compose.yml` above and under `/data/nginx-proxy-manager/app/data/logs` are multiple log files (`proxy_host-x.log`) files. I see zero data. Everything is 0. Just `Log Parsing Time` is 1 second. So I cannot see any old or live data. What am I doing wrong?

**Comment by fuuman** [comment]
Yes, I just hide the original path there. As mentioned I was in the goaccess container and was able to see and read the logs under `/opt/log`.

**Comment by xavier-hernandez** [comment]
Oh, you need to switch to the develop tag! The new code is only currently in the develop tag.

justsky/goaccess-for-nginxproxymanager:develop

**Comment by fuuman** [comment]
I am using the develop tag. Above is my quoted original post.

**Comment by Just5KY** [comment]
can you add root user perms in your compose and test

```yml
    environment:
        - PUID=0
        - PGID=0
```

**Comment by Just5KY** [comment]
I have updated `justsky/goaccess-for-nginxproxymanager:develop` image with the latest release.
if you can test it out @fuuman

**Comment by fuuman** [comment]
@Just5KY Just tested it. Still the same. 

Log: 
```
$ docker logs goaccess

GOAN v1.0.3

No archived logs found at /opt/log...
Checking active logs...

Processing (0) total log(s)...

 [SETTING UP STORAGE /goaccess/access.log] {0} @ {0/s}
WebSocket server ready to accept new client connections
```

Again I went into the container and checked `/opt/log`. Again all logs there. Readable.

If I can do anything else to help, please let me know.

**Comment by xavier-hernandez** [comment]
@fuuman just to summarize again. 

- you're using a Raspberry Pi?
   - If so, what version and OS
- can you post again your directory/files "ls -la", so we can see the user as well if possible?
- can you also post your docker-compose again for goaccess-for-nginxproxymanager?

Thanks.

**Comment by fuuman** [comment]
@xavier-hernandez 

OS: "Raspbian GNU/Linux 10 (buster)"

ls -la: 
```
# ls -la /opt/log
total 413972
drwxr-xr-x    2 root     root          4096 Mar 22 20:55 .
drwxr-xr-x    1 root     root          4096 Mar 23 20:53 ..
-rw-r--r--    1 root     root      18726057 Mar 22 20:19 default.log
-rw-r--r--    1 root     root       2105090 Mar 22 20:21 error.log
-rw-r--r--    1 root     root        115661 Mar 24 20:43 fallback_access.log
-rw-r--r--    1 root     root         48453 Mar 24 19:36 fallback_error.log
-rw-r--r--    1 root     root         12501 Nov  4 15:38 letsencrypt-requests.log
-rw-r--r--    1 root     root      54387247 Mar 24 20:49 proxy_host-1.log
-rw-r--r--    1 root     root      47349728 Mar 24 20:49 proxy_host-10.log
-rw-r--r--    1 root     root     195364576 Mar 23 18:53 proxy_host-11.log
-rw-r--r--    1 root     root             0 Jun 28  2021 proxy_host-12.log
-rw-r--r--    1 root     root       4341174 Mar 24 20:25 proxy_host-13.log
-rw-r--r--    1 root     root      26624337 Mar 24 20:48 proxy_host-14.log
-rw-r--r--    1 root     root         21056 Feb 19  2021 proxy_host-2.log
-rw-r--r--    1 root     root          1027 Feb 19  2021 proxy_host-3.log
-rw-r--r--    1 root     root      23906814 Mar 24 20:48 proxy_host-4.log
-rw-r--r--    1 root     root        275027 Nov 11 16:38 proxy_host-5.log
-rw-r--r--    1 root     root      48975360 Mar 24 20:49 proxy_host-6.log
-rw-r--r--    1 root     root        687386 Mar 24 00:07 proxy_host-7.log
-rw-r--r--    1 root     root        414662 Jun 16  2021 proxy_host-8.log
-rw-r--r--    1 root     root        445394 Mar 24 16:39 proxy_host-9.log
```

docker-compose.yml:
```
version: "3.9"
services:
  goaccess:
    image: justsky/goaccess-for-nginxproxymanager:develop
    container_name: goaccess
    restart: always
    environment:
        - TZ=Europe/Berlin
        - SKIP_ARCHIVED_LOGS=False #optional
        - BASIC_AUTH=False #optional
        - BASIC_AUTH_USERNAME=user #optional
        - BASIC_AUTH_PASSWORD=pass #optional
    ports:
        - '7880:7880'
    volumes:
        - /home/pi/docker_data/nginx-proxy-manager/app/data/logs:/opt/log
``

**Comment by xavier-hernandez** [comment]
@fuuman I have a PI finally up and running. I'll try checking on that one later today. The only difference I would see is that I'm running Raspbian GNU/Linux 11 (bullseye)

**Comment by xavier-hernandez** [comment]
:( I'm sorry but it's working for me. I have some sample logs that I use for testing. I even played with the owner of the files switching it from pi to root. Still worked.

```
total 11944
drwxr-xr-x 2 root root    4096 Mar 24 17:22 .
drwxr-xr-x 8 pi   pi      4096 Mar 24 17:25 ..
-rw-r--r-- 1 root root  249081 Mar 24 17:22 proxy-host-10_access.log.1.gz
-rw-r--r-- 1 root root 1039293 Mar 24 17:22 proxy-host-2_access.log
-rw-r--r-- 1 root root 1039293 Mar 24 17:22 proxy-host-2_error.log
-rw-r--r-- 1 root root  279719 Mar 24 17:22 proxy-host-2.log
-rw-r--r-- 1 root root 7230532 Mar 24 17:22 proxy-host-4_access.log
-rw-r--r-- 1 root root 2374124 Mar 24 17:22 redirection-host-2_access.log
pi@raspberrypi:~/sample-logs-develop $
```
``` yaml
version: "3.9"

services:
  goaccess:
    image: justsky/goaccess-for-nginxproxymanager:develop
    container_name: goaccess
    restart: always
    environment:
        - TZ=Europe/Berlin
        - SKIP_ARCHIVED_LOGS=False #optional
        - BASIC_AUTH=False #optional
        - BASIC_AUTH_USERNAME=user #optional
        - BASIC_AUTH_PASSWORD=pass #optional
    ports:
        - '7880:7880'
    volumes:
        - /home/pi/sample-logs-develop:/opt/log
```

![image](https://user-images.githubusercontent.com/6011223/160021177-b122ae3d-8e6e-4d61-995c-51c0ee853c58.png)

**Comment by fuuman** [comment]
Would it be possible that you give me access to your test files? I could copy them into my log folder. If it works, my log files from npm are trash. If it doesn't work my goaccess setup has a problem.

**Comment by xavier-hernandez** [comment]
Small sample files.
https://filebin.net/ae6p1g3o8p01gub5

**Comment by fuuman** [comment]
@xavier-hernandez Sorry,I had no time. Can you upload again? The filebin went down in the meantime.

**Comment by xavier-hernandez** [comment]
@fuuman 

I updated the develop image recently. Can you try that? I also added a DEBUG flag, check out the README file under the develop branch for more information. Also, an ARM version is available now from my docker repo.

docker pull xavierh/goaccess-for-nginxproxymanager:develop

**Comment by fuuman** [comment]
```
$ docker logs goaccess

GOAN v1.0.6

ls: /opt/log/proxy-host-*_access.log*.gz: No such file or directory
No archived logs found at /opt/log...
Checking active logs...
DEBUG - ON

Processing (0) total log(s)...

 [SETTING UP STORAGE /goaccess-config/access.log] {0} @ {0/s}
WebSocket server ready to accept new client connections
```
Which is correct because I have no archived access logs. My NPM does not generate any. I have just files named like "proxy_host-i.log"

**Comment by xavier-hernandez** [comment]
@fuuman sample files - https://filebin.net/u45b6p4i14uf5pt5

**Comment by fuuman** [comment]
@xavier-hernandez With these files everything is working correctly. 2 files downloaded into the log folder, 2 files imported automatically. 
I just have no idea why my NPM names the logfiles different than your NPM.

**Comment by xavier-hernandez** [comment]
@fuuman The files you downloaded included both versions of file names. The one you are using [proxy-host-2.log] and the one I'm using [proxy-host-4_access.log].

**Comment by fuuman** [comment]
No, I am using `proxy_host-2.log` with an underscore between `proxy` and `host`.

**Comment by xavier-hernandez** [comment]
@fuuman oh ok. I can modify the code for that then.

**Comment by xavier-hernandez** [comment]
can you open a separate issue for that?

**Comment by fuuman** [comment]
Would be awesome. Thank you :)

**Comment by fuuman** [comment]
Will be solved by #44.
